lib/PasswordStrategy.js

```diff
@@ -48,7 +48,8 @@ function Strategy(options) {
             sysPassword: password
           }, function(err, verified) {
             if(err) {
-              return callback(err);
+              // FIXME: Ignore errors to ensure all IDs are checked
+              return callback();
```

Comment:
Looking at #17, I'm guessing that the particular error here may have been due to an identity that has no password set. That seems like the kind of error that can be ignored if there is another identity that matches here. I think we have a few use cases here we need to cover:

1. A user is trying to verify the password for an identity that does have password-based authentication turned on and that identity is active in the system. It is conceivable that they could actually pass the check for at least this identity. There may be other identities with, for example, the same email address, that could never possibly pass due to `IdentityInactive` or `MethodNotAllowed` errors.

2. A user is trying to verify the password for an identity that is `IdentityInactive` or `MethodNotAllowed` and there is no other identity with the same email address that could possibly pass the check.

I think this means that we need to do two things:

1. Special case `IdentityInactive` and `MethodNotAllowed` errors by tracking them instead of reporting them, allowing all identity checks to be run.

2. If at least one identity reports back without an error, whether `verified` is `true` or `false`, ignore the above tracked errors and report valid/invalid password.

3. If no there is no such identity report an error indicating that there was no active identity with password authentication turned on for the given identifier.